doc/CodeStyle.md

```diff
@@ -0,0 +1,217 @@
+  * Declarations are done at or near their first use. If possible, declare iterators inside the `for` statement.
+  * Variables that contain commonly used objects often have the same name, try to use the same convention.
+  * Accesses to members of the class are *always* prefixed with `this->` to make them stand out with local variables.
+
+Simple statements like assignment or function calls that are very long can be split over multiple lines. In that case, add **2** additional TABs before the second and further lines to show the line is a continuation. For example
+```
+return this->titlebar.IsLoaded() && this->button.IsLoaded()
+		&& this->rounded_button.IsLoaded() && this->frame.IsLoaded()
+		...
+```
+
+Statements that have a block statement like `if`, `while`, `for`, and so on, may be either written at a single line, or split over sveral lines by adding curly brackets. The opening curly bracket is at the end of the line above the block, and the closing is at the same indentation as the starting indent, like
+```
+for (uint i = 0; i < TBN_COUNT; i++) this->bend_select[i] = NULL;
+
+while (true) {
+	if (iter == this->animations.end()) return;
+	...
+}
+```
+
+The `case` statement is indented inside the `switch` statement. To increase readability, an empty line is often useful between the alternatives. If you need local variables in the `case`, add them locally:
+```
+switch (wid_num) {
+	...
+
+	case BTB_SPACING: {
+		Point32 money = GetMoneyStringSize(LARGE_MONEY_AMOUNT);
+		...
+		break;
+	}
+
+	case ...:
+		...
+}
+```
+Normally, the switch is complete. Use the `NOT_REACHED();` macro to denote some cases will never happen. If a non-empty alternative falls through to the case below, use `/* FALL-THROUGH */` to denote this at the end of the former case.
+
+
+  * There is a space between the keyword (`if`, `while`, etc) and the opening parenthesis.
+  * Integer and pointer comparisons are explicit. That is, do `if (p != NULL)` or `while (i == 0)` instead of relying on the implicit conversion `if (p)` or `while (!i)`. Boolean comparisons are not explicit.
+  * `free` and `delete` can handle `NULL` values, no need to check against them yourself.
+  * A line containing only comment uses C-style comment, like `/* ... */`. Comments behind some code use line comment, like `a = 3; // ...`
+
+Other expression rules
+
+  * There are operator priorities in C++, you don't need to enforce them by adding parentheses. For uncommon or unclear cases, it may be useful to add them though.
+  * Use spaces around operators like `a + b` or `a != 4 && b * 2 < 15 * f`.
+
+Finally, as  a general rule, don't be over-generic in the implementation. Only add functionality in the functions and methods that you actually need now. If you need more tomorrow, add it tomorrow. Plans for the future sometimes don't happen, and then there is all this future functionality that nobody needs, which is only in the way.
```

Comment:
I think that phrase is correct English, but it could also be changed to, "which only gets in the way". Of course, the phrase could be entirely omitted and the same overall message would get across to the reader.

Actually, the following could work as well (IMO): "...that nobody needs and/or is hampered by."